sdk/core/core-xml/src/xml.browser.ts

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/// <reference lib="dom"/>
+
+const doc = document.implementation.createDocument(null, null, null);
+
+const parser = new DOMParser();
+export function parseXML(str: string, opts?: { includeRoot?: boolean }): Promise<any> {
+  try {
+    const dom = parser.parseFromString(str, "application/xml");
+    throwIfError(dom);
+
+    let obj;
+    if (opts && opts.includeRoot) {
+      obj = domToObject(dom);
+    } else {
+      obj = domToObject(dom.childNodes[0]);
+    }
+
+    return Promise.resolve(obj);
+  } catch (err) {
+    return Promise.reject(err);
+  }
+}
+
+let errorNS = "";
+try {
+  errorNS = parser.parseFromString("INVALID", "text/xml").getElementsByTagName("parsererror")[0]
```

Comment:
What's the goal of this block? as I understand, this is trying to figure the error structure, which may vary depending on the browser. Once that is figured out errorNS can be used to check if there are any errors.

Is the reason this is on a block by itself and not inside throwIfError, that we just want to run this (expensive?) figuring out once?

Reply:
Found the PR that introduced this line of code, but seems like it was grabbing the error NS before: https://github.com/Azure/ms-rest-js/pull/211/files

Setting the variable once has been in there since this support was originally added: https://github.com/Azure/ms-rest-js/pull/187

Perhaps @RikkiGibson could enlighten us?

Reply:
This was the best way I found at the time to discover the namespace used to report XML parsing errors, and therefore find out whether the return value of `parseFromString` represents a parse failure. I wrote it to discover the errorNS eagerly at startup because it's simple, and you need it every time you try to parse XML anyway. I can't speak to whether e.g. computing it from scratch each time or lazy initializing it is better.

Reply:
Thanks for the context, I think it's probably fine to do this eagerly, especially now that this code isn't even loaded as part of the main http pipeline and will only show up in services that need XML support.